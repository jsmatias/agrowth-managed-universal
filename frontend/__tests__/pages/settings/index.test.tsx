import React from 'react';
// tslint:disable-next-line:no-implicit-dependencies
import { render } from 'react-testing-library';
// tslint:disable-next-line:import-name
import SettingsPage from '../../../src/pages/settings/index';

// tslint:disable-next-line:no-console
console.error = jest.fn();

jest.mock('next/router', () => ({
  withRouter: jest.fn(() => () => (
    <h1 data-testid='withRouter'>Mocked withRouter</h1>
  ))
}));

describe('<SettingsPage/>', () => {
  test('should render', () => {
    const { queryByTestId } = render(<SettingsPage />);
    expect(queryByTestId('link-to-manage-produce')).toBeTruthy();
  });

  test('should not raise errors', () => {
    // tslint:disable-next-line:no-console
    expect(console.error).not.toBeCalled();
  });

  test('should match snapshot', () => {
    const { container } = render(<SettingsPage />);
    expect(container).toMatchSnapshot();
  });
});
