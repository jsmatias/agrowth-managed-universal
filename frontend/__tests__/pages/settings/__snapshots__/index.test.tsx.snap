// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsPage/> should match snapshot 1`] = `
<div>
  <h1
    data-testid="withRouter"
  >
    Mocked withRouter
  </h1>
  <ul>
    <li>
      <a
        data-testid="link-to-manage-produce"
        href="/settings/manage-produce"
      >
        Manage Produce
      </a>
    </li>
    <li>
      <a
        data-testid="link-to-manage-buyers"
        href="/settings/manage-buyers"
      >
        Manage Buyers
      </a>
    </li>
    <li>
      <a
        data-testid="link-to-manage-supplier"
        href="/settings/manage-supplier"
      >
        Manage Suppliers
      </a>
    </li>
  </ul>
</div>
`;
